### PR TITLE
Dependency upgrades

### PR DESCRIPTION
**Pull request by blushi**
## Description

Closes: regen-network/regen-registry#755, Closes: regen-network/regen-registry#769

Main packages upgrade:
- react
- typescript
- react-scripts to v4.0.3 (v5 just got released before the holidays and there are still some issues related to the usage of webpack 5 so I'd rather wait for the community to come up with proper solutions before upgrading: https://github.com/facebook/create-react-app/issues/11756) 
- react-router
- storybook
- mui

Gatsby will need to be updated once we upgrade react-scripts to v5 because otherwise that will cause conflicts with webpack versions in the monorepo (since react-scripts v4.0.3 still uses webpack 4)

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] provided a link to the relevant issue or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**.*

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles 
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by mhagel — APPROVED** [review]
Great work! Made a few comments but nothing very important to change. Thank you for this - we'll be in good shape for a while!

**Review by haveanicedavid — APPROVED** [review]
It seems like the majority of the diff is prettier changes so I haven't looked over every LOC, but pulling down your branch locally everything seems to run. 

I am noticing some style discrepancies, for ex the navbar items are showing a hover state:
Screen Shot 2022-01-19 at 6 11 18 PM

Different button sizes + image gradient on the /science page
Prod:
Screen Shot 2022-01-19 at 6 10 51 PM
image

This branch:

Screen Shot 2022-01-19 at 6 10 46 PM
image

etc, but I'm wondering if the best way to address those is by merging this PR first, then collectively (us + erika) itemizing necessary fixes like I listed above using the figma mockups as the source of truth which we could fix with a follow up PR.

WDYT? Should we try to correct any style issues before merging, or circle back for them after the merge is completed?

File: web-registry/src/pages/CreditsIssue.tsx [hunk 1]
```diff
@@ -226,30 +260,32 @@ function CreditsIssue(): JSX.Element {
         <div className={classes.datePicker}>
           <InputLabel id="project-select-label">Start Date</InputLabel>
           <DatePicker
-            autoOk
-            variant="inline"
+            // autoOk
```
[the remark this answers is not in the document]
Comment (blushi) [reply web-registry/src/pages/CreditsIssue.tsx]: yes the MUI pickers have been moved to lab and the API changed dramatically: https://mui.com/api/date-picker/
I removed the comments

**Comment by blushi** [comment]
> I am noticing some style discrepancies, for ex the navbar items are showing a hover state:

This is weird I can't reproduce it, could you try to describe precisely the steps to get that?

> Different button sizes + image gradient on the /science page

@haveanicedavid Fixed the button size, but could you be more specific about the difference with the image gradient? 

> WDYT? Should we try to correct any style issues before merging, or circle back for them after the merge is completed?


I think at this point, we can probably merge this and address any styles issues that might come up as separate bugs.

File: web-registry/src/App.css [hunk 2]
```diff
@@ -35,3 +35,8 @@
   margin: 0;
   -webkit-print-color-adjust: exact;
 }
+
+body {
+  font-family: 'Lato', -apple-system, sans-serif;
+  background-color: #fafafa;
```
[the remark this answers is not in the document]
Comment (blushi) [reply web-registry/src/App.css]: I've discussed this with @erikalogie and we came up with this follow-up issue https://github.com/regen-network/regen-registry/issues/803
set it to #fff as default for now

**Comment by haveanicedavid** [comment]
> could you be more specific about the difference with the image gradient?
@blushi sorry, I don't think I described that very well. I also don't think it's the gradient now that i'm looking at it, but if you compare those two images I shared - the first one has all of the text within the white area, while with the second the text bleeds into the trees a bit making it harder to read. 

> I think at this point, we can probably merge this and address any styles issues that might come up as separate bugs.
I agree 👍  lets get this merged asap

**Pull request merged**